### PR TITLE
Tag and Metadata features take an impl into param

### LINKED ISSUE #477: Make `{MetadataFeature, TagFeature}::new` take a `Into<Vec<u8>>`

### PR DESCRIPTION
**Pull request by thibault-martinez**
Fixes #477

### CODE REVIEW

**Review by qrayven — DISMISSED** [review]

File: sdk/src/types/block/output/feature/mod.rs [hunk 1]
```diff
@@ -431,10 +431,10 @@ pub mod dto {
                 FeatureDto::Sender(v) => Self::Sender(SenderFeature::new((&v.address).try_into()?)),
                 FeatureDto::Issuer(v) => Self::Issuer(IssuerFeature::new((&v.address).try_into()?)),
                 FeatureDto::Metadata(v) => Self::Metadata(MetadataFeature::new(
-                    prefix_hex::decode(&v.data).map_err(|_e| Error::InvalidField("MetadataFeature"))?,
+                    prefix_hex::decode::<Vec<u8>>(&v.data).map_err(|_e| Error::InvalidField("MetadataFeature"))?,
```
Comment (Alex6323) [inline sdk/src/types/block/output/feature/mod.rs:434]: why are we doing `_e`, and not just `_` if we don't care about the error?

File: sdk/src/wallet/account/operations/participation/voting.rs [hunk 2]
```diff
@@ -85,7 +85,7 @@ impl Account {
         let new_output = BasicOutputBuilder::from(output)
             // TODO maybe replace ?
```
Comment (Thoralf-M) [inline sdk/src/wallet/account/operations/participation/voting.rs]: Let's remove this, `with_features()` replaces them
```suggestion
```

**Review by Alex6323 — DISMISSED** [review]

File: sdk/src/wallet/account/operations/participation/voting.rs [hunk 3]
```diff
@@ -155,7 +155,7 @@ impl Account {
...
```
Comment (Thoralf-M) [inline sdk/src/wallet/account/operations/participation/voting.rs]: Same here 
```suggestion
```

File: sdk/src/types/block/output/feature/mod.rs (hunk 1 above)
Comment (thibault-martinez) [reply sdk/src/types/block/output/feature/mod.rs:434]: No specific reason

Comment (thibault-martinez) [reply sdk/src/types/block/output/feature/mod.rs:434]: Feel free to change later if you want, out of scope here

**Review by Thoralf-M — APPROVED** [review]

**Review by Alex6323 — APPROVED** [review]

**Pull request merged**